WeakAuras/Transmission.lua

```diff
@@ -730,283 +953,390 @@ local function scamCheck(codes, data)
-          trigger = data.additional_triggers[triggernum].trigger;
-        end
-        if(trigger) then
-          if(trigger.type == "aura") then
-            for index, name in pairs(trigger.names) do
-              local left = " ";
-              if(index == 1) then
-                if(#trigger.names > 0) then
-                  if(#trigger.names > 1) then
-                    left = L["Auras:"];
-                  else
-                    left = L["Aura:"];
-                  end
+      info[0], info.added = "added", 1
+    end
+    if children then
+      for i, child in ipairs(children) do
+        local oldChild = oldChildren[i]
+        if not oldChild then
+          modified = true
+          info[i], info.added = "added", info.added + 1
+        else
+          info.deleted[oldChild.id] = nil
+          info.deletedCount = info.deletedCount - 1
+          local childDiff= WeakAuras.diff(oldChild, child)
+          info[i] = {oldChild, childDiff}
+          if childDiff then
+            modified = true
+            info.modified = info.modified + 1
+            if tabsChanged < 7 then
+              for key in pairs(childDiff) do
+                local button = keyToButton[key]
+                if button and not info.tabsChanged[button] then
+                  info.tabsChanged[button] = true
+                  tabsChanged = tabsChanged + 1
                 end
+                if tabsChanged >= 7 then break end
               end
-
-              if (icons) then
-                tinsert(tooltip, {2, left, name..(icons[name] and (" |T"..icons[name]..":12:12:0:0:64:64:4:60:4:60|t") or ""), 1, 1, 1, 1, 1, 1});
-              else
-                local icon = WeakAuras.spellCache and WeakAuras.spellCache.GetIcon(name) or "Interface\\Icons\\INV_Misc_QuestionMark";
-                tinsert(tooltip, {2, left, name.." |T"..icon..":12:12:0:0:64:64:4:60:4:60|t", 1, 1, 1, 1, 1, 1});
-              end
-            end
-          elseif(trigger.type == "event" or trigger.type == "status") then
-            if(trigger.type == "event") then
-              tinsert(tooltip, {2, L["Trigger:"], (event_types[trigger.event] or L["Undefined"]), 1, 1, 1, 1, 1, 1});
-            else
-              tinsert(tooltip, {2, L["Trigger:"], (status_types[trigger.event] or L["Undefined"]), 1, 1, 1, 1, 1, 1});
-            end
-            if(trigger.event == "Combat Log" and trigger.subeventPrefix and trigger.subeventSuffix) then
-              tinsert(tooltip, {2, L["Message type:"], (WeakAuras.subevent_prefix_types[trigger.subeventPrefix] or L["Undefined"]).." "..(WeakAuras.subevent_suffix_types[trigger.subeventSuffix] or L["Undefined"]), 1, 1, 1, 1, 1, 1});
             end
           else
-            tinsert(tooltip, {2, L["Trigger:"], L["Custom"], 1, 1, 1, 1, 1, 1});
+            info.unmodified = info.unmodified + 1
           end
         end
       end
+      return (modified or next(info.deleted)) and info or false
```

Comment:
"next(info.deleted)" should be equivalent to "info.deletedCount > 0" which I find easier to understand.

Reply:
Holdover from an earlier iteration where I didn’t have deletedCount. I’ll change it.